packages/tabs/src/views/BottomTabBar.tsx

```diff
@@ -121,23 +122,37 @@ class TabBarBottom extends React.Component<BottomTabBarProps, State> {
     visible: new Animated.Value(1),
   };
 
+  keyboardDidShowSubscription: EmitterSubscription | null = null;
+  keyboardDidHideSubscription: EmitterSubscription | null = null;
+
   componentDidMount() {
     if (Platform.OS === 'ios') {
-      Keyboard.addListener('keyboardWillShow', this._handleKeyboardShow);
-      Keyboard.addListener('keyboardWillHide', this._handleKeyboardHide);
+      this.keyboardDidShowSubscription = Keyboard.addListener(
+        'keyboardWillShow',
+        this._handleKeyboardShow
+      );
+      this.keyboardDidHideSubscription = Keyboard.addListener(
+        'keyboardWillHide',
+        this._handleKeyboardHide
+      );
     } else {
-      Keyboard.addListener('keyboardDidShow', this._handleKeyboardShow);
-      Keyboard.addListener('keyboardDidHide', this._handleKeyboardHide);
+      this.keyboardDidShowSubscription = Keyboard.addListener(
+        'keyboardDidShow',
+        this._handleKeyboardShow
+      );
+      this.keyboardDidHideSubscription = Keyboard.addListener(
+        'keyboardDidHide',
+        this._handleKeyboardHide
+      );
     }
   }
 
   componentWillUnmount() {
-    if (Platform.OS === 'ios') {
-      Keyboard.removeListener('keyboardWillShow', this._handleKeyboardShow);
-      Keyboard.removeListener('keyboardWillHide', this._handleKeyboardHide);
-    } else {
-      Keyboard.removeListener('keyboardDidShow', this._handleKeyboardShow);
-      Keyboard.removeListener('keyboardDidHide', this._handleKeyboardHide);
+    if (this.keyboardDidShowSubscription) {
+      this.keyboardDidShowSubscription.remove();
```

Comment:
Should you reset the value to null here?

Reply:
@gauthierm i think we should remote the subscription when the component is unmount.

Reply:
I think he means you probably want to clear the variable here, and 3 lines below this one.
```suggestion
      this.keyboardDidShowSubscription.remove();
      this.keyboardDidShowSubscription = null;
```

Reply:
Isn't the variable garbage collected? And then re-initialized when the App opens again?

Reply:
Simplier:
```javascript
this.keyboardDidShowSubscription?.remove();
this.keyboardDidHideSubscription?.remove();
```